### PR TITLE
minor: unify it test code

### PR DESCRIPTION
**Pull request by rnveach**
@romani 
Here is an example of the start of some improvements on how we do tests in the IT folder.
If you agree, I will complete this. Should this go under 2540 or a new issue?

**Changes**:

ConfigurationBuilder was removed. Its needed parts were merged into BaseCheckTestSupport with some changes.
`getFilePath` was changed to be more like the Test folder, in that we will override the path and all inputs should be in the same package. This will remove us having issues finding files with duplicate names, and we remove the need for constantly scanning the folders for the files each instance.
`getConfigurationFromXML` was made static so we always have the google config ready at any time instead of recreating it each instance.
2 forms of `getCheckMessage` were merged into one.

OuterTypeFilenameTest is a working example of the changes that will be done to all the tests.

ConfigValidationTest was removed because of the changes to ConfigurationBuilder, but also because I think it is duplicating what all the IT tests do, which is run the code on a Checker to verify there are no issues with antlr.
I can restore this if you want.

### CODE REVIEW

**Comment by romani** [comment]
> Should this go under 2540 or a new issue?

too much refactoring for one issue, please create another issue with all these details.

> I can restore this if you want.

When you try to remove smth strange, please look at history of why this code is appeared
https://github.com/checkstyle/checkstyle/commits/master/src/it/java/com/google/checkstyle/test/base/ConfigValidationTest.java , it was recently created base on issue. We need to keep it.

> If you agree, I will complete this

Your intention looks reasonable for me, @mkordas please review share your opinion.

**Comment by rnveach** [comment]
> Please take a look at latest code

Yes, this was made before your changes, I saw the conflicts on rebase and fixed them.

> We do not have checkstyle_config.xml enforced over ITs ?

We do, what violation are you expecting? Parameter name check uses `"^[a-z][a-zA-Z0-9]+$"` which makes `aConfig` valid. We also have variables like this in test.
https://github.com/rnveach/checkstyle/blob/189da7314d1b29f0e14479210c2fd567305a64d2/src/test/java/com/puppycrawl/tools/checkstyle/BaseCheckTestSupport.java#L119

> please look at history... We need to keep it.

Sorry, I fixed it.

> please create another issue with all these details

Done. #2589 

**Additional Changes**:
It is ready for final review. Split into 3 commits.
Commit 1 contains the changes to the base code.
Commit 2 contains fixes like inputs not being in the same package as the test, tests originally being run on inputs from different packages, tests not having their own package input file.
Commit 3 removes excess warning comments that aren't warnings anymore and split input files into 2 if same package has to run 2 different tests on the file (each one produces a different set of warnings).

File: src/it/java/com/google/checkstyle/test/base/BaseCheckTestSupport.java [hunk 1]
```diff
@@ -66,114 +90,86 @@ protected Checker createChecker(Configuration aCheckConfig)
...
      */
     protected String getCheckMessage(Class<? extends AbstractViolationReporter> aClass,
-            String messageKey) {
+            String messageKey, Object... arguments) {
         final Properties pr = new Properties();
         try {
             pr.load(aClass.getResourceAsStream("messages.properties"));
         }
-        catch (IOException ignored) {
+        catch (IOException e) {
```
Comment (mkordas) [inline src/it/java/com/google/checkstyle/test/base/BaseCheckTestSupport.java]: What is the reason for this change?

**Comment by mkordas** [comment]
> @mkordas please review share your opinion.

Approach is good, I like it. But I think that there are far too many changes going on especially in commit  1, so it is very hard to review it in reasonable time.

**Comment by rnveach** [comment]
@mkordas 

> What is the reason for this change?

Test's `BaseCheckTestSupport` had a similar method and named it's catch "ignored". Originally, this method in IT was 2 different methods, so I copied the method from Test's (since it only needed one implementation) keeping parts of IT's unique implementation.
I did a lot of comparing between IT and Test's implementation of `BaseCheckTestSupport` for my changes. Hence why I named the first commit "Unify IT code", as I am making IT and Test coded similarly.
I can always fix this line if it is needed.

> This rule is named "package statement", why this line is changed?

The test is in package `rule32packagestate` and the input was in `rule32packagestatement`.
Since they should be in the same package, I always went with the test as the correct package since it is part of the URL in xdoc.
This also happened with `rule3sourcefilestructure` and `rule3sourcefile`, and `rule62caughtexceptions` and `rule62donotignoreexceptions`.
I can fix if we wish to go with the other, or a completely different, name.

**Comment by romani** [comment]
merged as FF
